x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeListener.java

```diff
@@ -71,19 +175,56 @@ ActionListener<Void> acquireAvoid() {
     }
 
     /**
-     * Acquires a new listener that collects compute result. This listener will also collects warnings emitted during compute
+     * Acquires a new listener that collects compute result. This listener will also collect warnings emitted during compute
+     * @param computeClusterAlias The cluster alias where the compute is happening. Used when metadata needs to be gathered
+     *                            into the {@link EsqlExecutionInfo} Cluster objects.
      */
-    ActionListener<ComputeResponse> acquireCompute() {
+    ActionListener<ComputeResponse> acquireCompute(@Nullable String computeClusterAlias) {
```

Comment:
Could you add an explanation to the javadoc about when this is `null` vs the special local key?

Reply:
Looks like when it's null we don't gather any metadata at al.

Reply:
Good idea. Added in next push.